varada_trino_manager/infra/call_home_methods.py

```diff
@@ -0,0 +1,242 @@
+        prefix = self.path.split("*")[0]
+        bucket = self.bucket
+        paginator = client.get_paginator("list_objects_v2")
+        for page in paginator.paginate(Bucket=bucket, Prefix=prefix):
+            for obj in page.get("Contents", []):
+                if obj["StorageClass"] == "STANDARD" and fnmatch.fnmatch(obj["Key"], self.path):
+                    path = obj["Key"]
+                    url = f's3://{bucket}/{path}'
+                    yield S3URL(url=url, etag=obj["ETag"])
+
+    def glob_folders(self):
+        prefix = self.path.split("*")[0]
+        bucket = self.bucket
+        paginator = client.get_paginator("list_objects_v2")
+        root_url = str(self.url) + '/'
+
+        for page in paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter='/'):
+            for obj in page.get("CommonPrefixes", []):
+                path = obj["Prefix"]
+                url = f'{self.url.scheme}://{bucket}/{path}'
+                if root_url == url:  # ignore the root directory
+                    continue
+                yield S3URL(url=url)
+
+
+def get_slog_metrics(slof_files: list, frequency_minutes: int, start_time, end_time, delta_metrics):
+    catalog_ts_jsons = collections.defaultdict(lambda: collections.defaultdict(dict))
+    for slog in slof_files:
+        for line in slog.split('\n'):
+            if "METRICS-DUMP" in line:
+                line_splits = re_findall(r'([\S]+)[\s]+INFO.*METRICS-DUMP[\s]+(.*)', line)
+                if len(line_splits[0]) > 0 and "stats\":" in line_splits[0][1]:
+                    json_string = line_splits[0][1]
+                    json_data = json.loads(json_string)
+                    timestamp = json_data.get('timestamp', line_splits[0][0])/1000  # seconds
+                    timestamp_str = datetime.datetime.fromtimestamp(timestamp).strftime('%m/%d/%Y %H:%M')
+
+                    if timestamp < start_time or timestamp > end_time:
+                        print(f" {timestamp_str} not in range")
+                        continue
+                    timestamp = int(timestamp / (60 * frequency_minutes))  # sum_minutes
+                    catalog = json_data.get('catalog', 'varada')
+
+                    stats: dict = json_data['stats']
+                    for group_tuple in stats.items():
+                        group_name_with_catalog = group_tuple[0]
+                        group_name_splits = re_findall(r"([^\.]+)", group_name_with_catalog)
+                        group_name = group_name_splits[0]
+                        metrics = dict(group_tuple[1]).items()
+                        for metric_tuple in metrics:
+                            metric_name = metric_tuple[0]
+                            metric_value = metric_tuple[1]
+                            values = re_findall(r"([\+\-][\d]+).*\(([\d]+)\)", metric_value)
+
+                            if metric_name in delta_metrics:
+                                diff = int(values[0][1])
```

Comment:
You're doing this access many times `x[0][1]` consider adding a helper function
```
def get(obj):
    return obj[0][1]
```
Don't forget to add input checks and error handling